### DISCUSSION TITLE
How can I merge two of my github accounts

### DISCUSSION THREAD

**Discussion by teleansh** [community/community#49976] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I had a personal GitHub account but I recently create a student's account from my student email id to get the benefits, is there any way that I can merge these accounts?
Thanks

**Comment by Fb9027** [comment]
I have somebody trying to log into my Facebook Instagram

**Comment by Anmol-Baranwal** [comment, marked as answer]
> I had a personal GitHub account but I recently create a student's account from my student email id to get the benefits, is there any way that I can merge these accounts? Thanks

Yes, it is possible to merge your personal GitHub account with your student account. GitHub allows you to transfer your personal repositories and settings to your student account, and then close your personal account.

Here are the steps to merge your accounts:

1. Make sure that your student account has the same email address associated with it as your personal account.

2. Go to your personal account's settings page and scroll down to the "Danger Zone" section.

3. Click on the "Transfer account" button and follow the prompts to transfer your personal repositories and settings to your student account.

4. Once the transfer is complete, log out of your personal account and log in to your student account.

5. Go to your student account's settings page and confirm that your repositories and settings have been transferred.

6. If everything looks good, you can close your personal account by going to the "Danger Zone" section of your personal account's settings page and clicking on the "Close account" button.

Note that when you transfer your personal account to your student account, your commit history and other personal data will not be transferred. Additionally, you will need to update any Git remotes that point to your personal account's repositories to point to your student account's repositories.

**Reply by Anmol-Baranwal** [reply]
@teleansh 
I just wanted to check in and see if you had any further questions or concerns about my answer on GitHub. If you have no further doubts or if your doubts have been resolved, would you mind marking my answer as accepted?

It would be greatly appreciated and it will also help other users who might be looking for a solution to a similar issue.

**Reply by ghost** [reply]
I can't find "Danger Zone". Was it moved?

**Reply by greatvic** [reply]
what happens when your student account is closed, after graduation?

Does it resort to your personal e-mail as the primary account holder automatically?

**Reply by nieladrian-prog** [reply]
Actually, GitHub doesn’t have a “Transfer account” button in the Danger Zone. What you can do is manually merge accounts by:
Adding both emails to the account you want to keep
Transferring repositories one by one through repo settings → Danger Zone → “Transfer ownership”
Re-adding SSH keys, PATs, etc.
For GitHub Education benefits, they don’t move automatically — you need to request a transfer from GitHub Education Support

**Comment by Ntshangase** [comment]
Hello, I don't think you had to create a new account to get the student benefits. I also have a personal account but when I registered for the student benefits I just used the student email( when requested for email ) and GitHub gave me an option to make my student account a secondary email, which allowed me to keep my original personal GitHub account but still get the benefits like GitHub Global Campus.

You can learn more here: https://docs.github.com/en/education/explore-the-benefits-of-teaching-and-learning-with-github-education/github-global-campus-for-students/apply-to-github-global-campus-as-a-student

For assistance I would suggest you log an issue here (to merge account) : https://support.github.com/contact/education?tags=hubberfy_education

**Reply by nieladrian-prog** [reply]
That’s true if you apply the first time using your personal account and just add the school email as secondary.
But in my case, I already used my school email on a separate GitHub account before, so the Education benefits are locked to that account. Now when I try to apply on my personal account with the same school email, GitHub blocks it since it’s already been used once.

**Comment by ghost** [comment]
This is no help. GitHub will not let me use either of my accounts' primary email addresses to merge to one another. What solution is there?
